### DISCUSSION TITLE
Pre-commit

### DISCUSSION THREAD

**Discussion by mbway** [mozman/ezdxf#879] [Ideas]
since I think the code is formatted with black and type checked with mypy it might be a good idea to have a `.pre-commit-config.yaml` (see: [pre-commit](https://pre-commit.com/)) at the root of the repo to make it easy for contributors to make sure they are following the guidelines.

**Comment by mozman** [comment]
I'm glad the Github workflows work the way they are. I don't want to add more complexities that I might not understand if they break in six months.

**Reply by mbway** [reply]
pre-commit is a tool optionally run locally before making each commit, so contributors could catch issues before reaching github.
It's quite a popular tool. For example of the ezdxf dependencies: matplotlib and pytest use it in their repos
